components/PriceGraph/AuctionHousePriceGraph/AuctionHousePriceGraph.tsx

```
@@ -69,6 +93,83 @@ function AuctionHousePriceGraph(props: Props) {
         })
     }, [props.item.tag])
 
+    useEffect(() => {
+        // Check premium status
+        if (typeof window !== 'undefined' && sessionStorage.getItem('googleId')) {
+            api.getPremiumProducts().then(products => {
+                setHasPremium(hasHighEnoughPremium(products, PREMIUM_RANK.PREMIUM))
+            }).catch(() => {
+                setHasPremium(false)
+            })
+        }
+    }, [])
+
+    function handleAfterLoginForPremium() {
+        // Re-check premium products after login and retry year chart if allowed
+        api.getPremiumProducts().then(products => {
+            const ok = hasHighEnoughPremium(products, PREMIUM_RANK.PREMIUM)
+            setHasPremium(ok)
+            if (ok) {
+                updateChart(DateRange.YEAR, itemFilter)
+            }
+        }).catch(() => {})
```

sad empty catch block

---

that exists 3 times in the project already